app/src/main/java/com/example/tasky/common/presentation/TextBox.kt

```diff
@@ -14,32 +16,76 @@ import androidx.compose.runtime.remember
 import androidx.compose.runtime.setValue
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.colorResource
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.text.input.PasswordVisualTransformation
+import androidx.compose.ui.text.input.VisualTransformation
 import androidx.compose.ui.tooling.preview.Preview
 import androidx.compose.ui.unit.dp
+import com.example.tasky.R
+import com.example.tasky.common.domain.isValidName
 
 @Composable
-fun TextBox(hintText: String) {
-    var text by remember { mutableStateOf("") }
+fun TextBox(
+    hintText: String,
+    text: String,
+    onValueChange: (String) -> Unit,
+    validator: (String) -> Boolean,
+    isPasswordField: Boolean = false
+) {
+    var isValid by remember { mutableStateOf(validator(text)) }
+    var passwordVisible by remember { mutableStateOf(false) }
 
     TextField(
         value = text,
-        onValueChange = { text = it },
+        onValueChange = {
+            onValueChange(it)
+            isValid = validator(it)
+        },
         modifier = Modifier
             .fillMaxWidth()
             .padding(top = 16.dp, start = 16.dp, end = 16.dp),
         shape = RoundedCornerShape(10.dp),
         colors = TextFieldDefaults.colors(
-            focusedIndicatorColor = Color.Transparent, // Set focused indicator color to match background
-            unfocusedIndicatorColor = Color.Transparent // Set unfocused indicator color to match background
+            focusedIndicatorColor = Color.Transparent,
+            unfocusedIndicatorColor = Color.Transparent
         ),
-        label = { Text(hintText) }
+        label = { Text(hintText) },
+        trailingIcon = {
+            if (isValid) {
+                Icon(
+                    painter = painterResource(id = R.drawable.ic_green_checkmark),
+                    contentDescription = "Valid",
+                    tint = colorResource(id = R.color.tasky_green)
+                )
+            } else if (isPasswordField) {
+                val image = if (passwordVisible) {
+                    painterResource(id = R.drawable.ic_password_show)
+                } else {
+                    painterResource(id = R.drawable.ic_password_hide)
+                }
+                IconButton(onClick = { passwordVisible = !passwordVisible }) {
+                    Icon(
+                        painter = image,
+                        contentDescription = if (passwordVisible) "Hide password" else "Show password",
```

Comment:
If you're using string resources all over the place, it also makes sense to do that for content descriptions, since someone who would rely on a screen reader (for whom content descriptions would become important) would also expect them to be in their local language.